smoke-test/src/test/java/org/opennms/smoketest/flow/BmpIT.java

```diff
@@ -167,6 +168,11 @@ public void verifyBmpProcessing() {
                 .then().assertThat()
                 .statusCode(201);
 
+        // Restarting collectd after we have added a node to the db
```

Comment:
There is no need to restart `collectd`  every time we add node to DB.  `Collectd` need to be restarted only if collectd configuration changes.

Reply:
During the debug of the test I found that if we add a node before collectd initialized, the test will pass. However, if we add a node after, we need to wait for 5 min for the next collectd round in order to get RRD file not empty. 
Instead of making this test 6 min longer, Ben and I agreed to restart collectd to make a test stabel.

Reply:
Not sure how it is related to `Collectd`.  We are directly trying to get Resources from ResourceRestService which  maps resources from file system.  

From the logs [here](https://circleci-tasks-prod.s3.us-east-1.amazonaws.com/storage/artifacts/670a6d48-e2c1-4c1a-ab60-a1ad8faefbda/51758ee6-b885-46de-b28f-844e4a6000eb/2/test-artifacts/logs/org.opennms.smoketest.flow.BmpIT-null/opennms/karaf.log?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQVFQINEOEE5AS5VR%2F20220722%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20220722T131046Z&X-Amz-Expires=60&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEBUaCXVzLWVhc3QtMSJHMEUCIQClxTor2KP7pRi0SSVOGxFGlfuCFHaF0423TBX5qPUYFQIgQJ0K9h%2BTdPk1NlJELBAPUL0lH7hofSU0wwImBPYgElgqqwIIXhADGgwwNDU0NjY4MDY1NTYiDB7fPIjroNsdJi1g9CqIAuUp1iqluLcOqfjXPUpTd3o5aWdBbaxLRdklLbBvPsSWf%2F7ZHpnHElpHRHHmEPSw0cBPNBbUeeTAfRyJFqIQvW%2BQVUguc5TPQfEAQSmbusreCfhbCuwfy3S3UspX5FmQEWBYiEF6ShJvtIQhdmtuP%2BcJkl9EaNUN5LgOd0u93L3hYV725Sb3Kg73nzgGbThTnAM0txdaFf4occi1TMBjbQvLbiYTG%2FSV%2FunBSmJ7PnJG3f2f4ZbO38ZkYGwNBC96Z5alYyy4ULdcqlwHVkOTRI3fXG9VWRQ84anmQWDs7mV2a0BQ7860vKp3wFIWbypJ0%2ByJbSHlsOxCB0%2BKRIC6Xir0ihAa%2FVUSojC0veqWBjqdAbBwgdgNgPD4UNdnpOtsAl%2BZJCxWvUDTSv2Z2T3LgJ3IMVDvVdRoB4jugPxfjf4UJbJV57cp5%2BDB5Z1vVWsf16JBZXS0xdR4p5he%2BM0hRd33RByWW3X%2B6PPyz%2Bsmz8M7seJoZczkw3PFTv%2FPiSCHNBiRJt2FgiLM6ndZS0YBzEdgIlANM%2Fw4oATHn2BX%2F4SOUmv3B116oECMVZnoh7Y%3D&X-Amz-SignedHeaders=host&x-id=GetObject&X-Amz-Signature=016b06e9a675d735425278a548e8563bb48f426e59f7308a3588ba2b5fcf1ce9), it looks like `BmpAdapter` can't find the node it needs to export resources.  

In any case, it will be good to add a comment why we are doing this workaround.

Reply:
@cgorantla

We spent some time pairing on this yesterday, and as far as we could tell, it's not so much that collectd needs a restart because of nodes, it's that there is a race condition between collectd starting, and the test adding new nodes to be part of collection scanning. What we found was (I think):

### passing test
1. "opennms finished loading" is returned, so the test starts executing
2. collectd has _not_ started its first scan yet
3. the test writes a node to the database
4. collectd starts its first scan, telemetry scanning is triggered for the BMP test node
5. test checks the filesystem and sees collection data written

### failing test
1. "opennms finished loading" is returned, so the test starts executing
2. collectd runs its first scan, sees only the localhost node, nothing BMP-related happens
3. collectd executor starts its 5 minute timer to wait for the next collection run
4. test writes a node to the database
5. test checks the filesystem and does not see collection data written

It's a race condition between collectd's first collection, and whether that BMP node got written in time. The test could be fixed by using a custom collectd config that scans more often and then waiting a longer period of time, but it's easier to just let collectd start over as soon as we know the node is in the database, and then the test will complete quickly.

Alternately: I know nothing of how telemetry is triggered and this just happens to be a lucky side-effect. :D

Honestly it seems weird to me that the test is making sure BMP is being processed by checking an RRD file, rather than the elasticsearch cluster that is started alongside the test. But I don't know the guts of this stuff to make it do that. (Perhaps really, it should be checking both.)

Reply:
This is BMP Telemetry Adapter which uses `AbstractCollectionAdapter` to push metrics to RRD directly outside `Collectd` scan phase.  So it shouldn't be related unless there is some sync on Resources associated with Collectd scan phase.   

It is also worth looking  for this [failure]( https://circleci-tasks-prod.s3.us-east-1.amazonaws.com/storage/artifacts/670a6d48-e2c1-4c1a-ab60-a1ad8faefbda/51758ee6-b885-46de-b28f-844e4a6000eb/2/test-artifacts/logs/org.opennms.smoketest.flow.BmpIT-null/opennms/karaf.log?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQVFQINEOEE5AS5VR%2F20220722%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20220722T131046Z&X-Amz-Expires=60&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEBUaCXVzLWVhc3QtMSJHMEUCIQClxTor2KP7pRi0SSVOGxFGlfuCFHaF0423TBX5qPUYFQIgQJ0K9h%2BTdPk1NlJELBAPUL0lH7hofSU0wwImBPYgElgqqwIIXhADGgwwNDU0NjY4MDY1NTYiDB7fPIjroNsdJi1g9CqIAuUp1iqluLcOqfjXPUpTd3o5aWdBbaxLRdklLbBvPsSWf%2F7ZHpnHElpHRHHmEPSw0cBPNBbUeeTAfRyJFqIQvW%2BQVUguc5TPQfEAQSmbusreCfhbCuwfy3S3UspX5FmQEWBYiEF6ShJvtIQhdmtuP%2BcJkl9EaNUN5LgOd0u93L3hYV725Sb3Kg73nzgGbThTnAM0txdaFf4occi1TMBjbQvLbiYTG%2FSV%2FunBSmJ7PnJG3f2f4ZbO38ZkYGwNBC96Z5alYyy4ULdcqlwHVkOTRI3fXG9VWRQ84anmQWDs7mV2a0BQ7860vKp3wFIWbypJ0%2ByJbSHlsOxCB0%2BKRIC6Xir0ihAa%2FVUSojC0veqWBjqdAbBwgdgNgPD4UNdnpOtsAl%2BZJCxWvUDTSv2Z2T3LgJ3IMVDvVdRoB4jugPxfjf4UJbJV57cp5%2BDB5Z1vVWsf16JBZXS0xdR4p5he%2BM0hRd33RByWW3X%2B6PPyz%2Bsmz8M7seJoZczkw3PFTv%2FPiSCHNBiRJt2FgiLM6ndZS0YBzEdgIlANM%2Fw4oATHn2BX%2F4SOUmv3B116oECMVZnoh7Y%3D&X-Amz-SignedHeaders=host&x-id=GetObject&X-Amz-Signature=016b06e9a675d735425278a548e8563bb48f426e59f7308a3588ba2b5fcf1ce9) on same test where BMP Adapter doesn't find node while pushing metrics.  We may have to wait till we find node & Interface ( by querying Dao) and then send BMP packets.